File: clonewhatapp/CloneWhatsApp.API/DTOs/MensajeDTOs.cs
namespace CloneWhatsApp.API.DTOs;

public class MensajeDTO
{
    public Guid Id { get; set; }
    public Guid ChatId { get; set; }
    public Guid RemitenteId { get; set; }
    public string? RemitenteNombre { get; set; }
    public string? Contenido { get; set; }
    public string Tipo { get; set; } = string.Empty;
    public string? UrlArchivo { get; set; }
    public string? NombreArchivo { get; set; }
    public long? TamanoArchivo { get; set; }
    public int? DuracionSegundos { get; set; }
    public Guid? MensajeRespondidoId { get; set; }
    public MensajeDTO? MensajeRespondido { get; set; }
    public DateTime FechaEnvio { get; set; }
    public bool Eliminado { get; set; }
    public bool EliminadoParaTodos { get; set; }
    public bool Editado { get; set; }
    public string Estado { get; set; } = string.Empty; // Enviado, Entregado, Leido
}

public class EnviarMensajeDTO
{
    public Guid ChatId { get; set; }
    public string? Contenido { get; set; }
    public string Tipo { get; set; } = "Texto";
    public Guid? MensajeRespondidoId { get; set; }
}

public class MensajeEnviadoDTO
{
    public Guid MensajeId { get; set; }
    public Guid ChatId { get; set; }
    public MensajeDTO Mensaje { get; set; } = null!;
}

public class ActualizarEstadoMensajeDTO
{
    public Guid MensajeId { get; set; }
    public string Estado { get; set; } = string.Empty; // Entregado, Leido
}

public class TypingDTO
{
    public Guid ChatId { get; set; }
    public Guid UsuarioId { get; set; }
    public string NombreUsuario { get; set; } = string.Empty;
    public bool EstaEscribiendo { get; set; }
}

public class ForwardMessageDTO
{
    public Guid TargetChatId { get; set; }
}

File: clonewhatapp/whatsapp-clone-android/core/network/src/main/java/com/clonewhatsapp/core/network/dto/MensajeDtos.kt
package com.clonewhatsapp.core.network.dto

import com.google.gson.annotations.SerializedName

/**
 * DTO for message data from backend
 * Matches backend MensajeDTO
 */
data class MensajeDto(
    @SerializedName("Id")
    val id: String,

    @SerializedName("ChatId")
    val chatId: String,

    @SerializedName("RemitenteId")
    val remitenteId: String,

    @SerializedName("RemitenteNombre")
    val remitenteNombre: String?,

    @SerializedName("Contenido")
    val contenido: String?,

    @SerializedName("Tipo")
    val tipo: String,

    @SerializedName("UrlArchivo")
    val urlArchivo: String?,

    @SerializedName("NombreArchivo")
    val nombreArchivo: String?,

    @SerializedName("TamanoArchivo")
    val tamanoArchivo: Long?,

    @SerializedName("DuracionSegundos")
    val duracionSegundos: Int?,

    @SerializedName("MensajeRespondidoId")
    val mensajeRespondidoId: String?,

    @SerializedName("MensajeRespondido")
    val mensajeRespondido: MensajeDto?,

    @SerializedName("FechaEnvio")
    val fechaEnvio: String,

    @SerializedName("Eliminado")
    val eliminado: Boolean,

    @SerializedName("EliminadoParaTodos")
    val eliminadoParaTodos: Boolean,

    @SerializedName("Editado")
    val editado: Boolean,

    @SerializedName("Estado")
    val estado: String // Enviado, Entregado, Leido
)

/**
 * DTO for sending a text message
 * Matches backend EnviarMensajeDTO
 */
data class EnviarMensajeRequest(
    @SerializedName("ChatId")
    val chatId: String,

    @SerializedName("Contenido")
    val contenido: String?,

    @SerializedName("Tipo")
    val tipo: String = "Texto",

    @SerializedName("MensajeRespondidoId")
    val mensajeRespondidoId: String?
)

/**
 * DTO for forwarding a message
 * Matches backend ForwardMessageDTO
 */
data class ForwardMessageRequest(
    @SerializedName("TargetChatId")
    val targetChatId: String
)
